code/modules/shuttle/navigation_computer.dm

```diff
@@ -26,19 +26,15 @@
 	GLOB.navigation_computers += src
 
 	if(!mapload)
-		connect_to_shuttle(SSshuttle.get_containing_shuttle(src))
+		connect_to_shuttle(SSshuttle.get_containing_shuttle(src), SSshuttle.get_containing_dock(src))
 
-		for(var/port_id in SSshuttle.stationary)
-			var/obj/docking_port/stationary/S = SSshuttle.stationary[port_id]
-			if(S.id == shuttleId)
+		for(var/port in SSshuttle.stationary)
+			if(!port)
+				continue
+			var/obj/docking_port/stationary/S = port
+			if(S.port_destinations == shuttleId)
 				jumpto_ports[S.id] = TRUE
 
-	for(var/V in SSshuttle.stationary)
-		if(!V)
-			continue
-		var/obj/docking_port/stationary/S = V
-		if(jumpto_ports[S.id])
-			z_lock |= S.z
```

Comment:
What did this code do and why have you removed it?

Reply:
This part of code collect to `z_lock` z levels of initial ports available to jump